lib/rack/oauth2/server/token.rb

```diff
@@ -56,6 +56,15 @@ def initialize(env)
             else
               super
               @client_secret = params['client_secret']
+              @client_assertion = params['client_assertion']
+              @client_assertion_type = params['client_assertion_type']
+              if client_assertion.present? && client_assertion_type == URN::ClientAssertionType::JWT_BEARER
+                require 'json/jwt'
```

Comment:
Should we always require this? I don't know what the performance implications are of doing it here as opposed to always.

Reply:
JWT isn't used in general OAuth use-cases.
It's required only when JWT assertion use-case.